src/sqlfluff/core/rules/base.py

```diff
@@ -309,6 +310,8 @@ def crawl(
                 path=pathlib.Path(fname) if fname else None,
                 templated_file=templated_file,
             )
+        except (bdb.BdbQuit, KeyboardInterrupt):
```

Comment:
Unrelated to the overall issue, but this makes SQLFluff better behaved: If a user or developer hits ^C at runtime or quits out of the debugger, don't catch the exception.

Reply:
```suggestion
        except (bdb.BdbQuit, KeyboardInterrupt):  #pragma: no cover
```